### PR TITLE
Send notifications for new keeps to relevant libraries/users

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@leogrim @camhashemi

### CODE REVIEW

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala [hunk 1]
```diff
@@ -201,6 +176,26 @@ class ElizaDiscussionCommanderImpl @Inject() (
...
+        val mt = messageThreadRepo.intern(MessageThread(
+          uriId = csKeep.uriId,
+          url = csKeep.url,
+          nUrl = csKeep.url,
+          pageTitle = csKeep.title,
+          startedBy = csKeep.owner getOrElse owner,
+          participants = MessageThreadParticipants(users),
+          keepId = csKeep.id,
+          numMessages = 0
+        ))
```
Comment (leogrim) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala:192]: `messageThreadRepo.intern` will do another look up here, should we have this repo method return the `Boolean`?

File: kifi-backend/modules/common/app/com/keepit/common/routes/Router.scala [hunk 2]
```diff
@@ -277,6 +277,7 @@ object Eliza extends Service {
     def convertNonUserThreadToUserThread(userId: Id[User], accessToken: String) = ServiceRoute(POST, "/internal/eliza/convertNonUserThreadToUserThread", Param("userId", userId), Param("accessToken", accessToken))
 
     def handleKeepEvent = ServiceRoute(POST, "/internal/eliza/handleKeepEvent")
+    def internEmptyThreads = ServiceRoute(POST, "/internal/eliza/internEmptyThreads")
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/common/routes/Router.scala]: `internThreadForKeeps` to be consistent with the other end?

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala [hunk 3]
```diff
@@ -24,6 +24,7 @@ import scala.concurrent.{ ExecutionContext, Future }
 
 @ImplementedBy(classOf[ElizaDiscussionCommanderImpl])
 trait ElizaDiscussionCommander {
+  def internThreadForKeep(csKeep: CrossServiceKeep, owner: Id[User]): (MessageThread, Boolean)
```
Comment (leogrim) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala:27]: Do we need the additional `owner` here? I haven't tracked whether it's different from `csKeep.owner`

Comment (ryanpbrewster) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala:27]: Threads have a non-optional owner

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala (hunk 1 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala:192]: seems reasonable

File: kifi-backend/modules/common/app/com/keepit/common/routes/Router.scala (hunk 2 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/common/routes/Router.scala]: seems reasonable

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala (hunk 3 above)
Comment (leogrim) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala:27]: What's the logic to pick the owner? First `Keep.user` or oldest `KeepToUser`? Do you know what happens with re-attribution if there's already a `MessageThread`? Perhaps we just want to kill thread owners?

Comment (leogrim) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala:27]: (I think I remember you wanting to kill `MessageThread` altogether)

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala (hunk 1 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala:192]: looks like `intern` is poorly named, it assumes that there is no active model

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala (hunk 3 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala:27]: The keep owner, if there is one. If not, the first user to take an action on the keep.

**Pull request merged**